### PR TITLE
expose rest api for splits

### PR DESCRIPTION
**Pull request by JerryKwan**
expose rest api for split list
change rest api response data structure
add offset/limit pagniation
change SplitClient list() accordingly

### Description
Solve the issues described in https://github.com/quickwit-oss/quickwit/issues/3907

### How was this PR tested?

Currently it supports the operations like this
curl  "http://127.0.0.1:7280/api/v1/indexes/stackoverflow/splits?offset=10&limit=10"
and this
./quickwit split list --index stackoverflow --offset 10

And the response of the rest api looks like the following
```
{
  "offset": 0,
  "size": 1,
  "splits": [
    {
      "split_state": "Published",
      "update_timestamp": 1695642901,
      "publish_timestamp": 1695642901,
      "version": "0.6",
      "split_id": "01HB632HD8W6WHNM7CZFH3KG1X",
      "index_uid": "stackoverflow:01HB6321TDT3SP58D4EZP14KSX",
      "partition_id": 0,
      "source_id": "_ingest-api-source",
      "node_id": "jerry",
      "num_docs": 10000,
      "uncompressed_docs_size_in_bytes": 6674940,
      "time_range": {
        "start": 1217540572,
        "end": 1219335682
      },
      "create_timestamp": 1695642900,
      "maturity": {
        "type": "immature",
        "maturation_period_millis": 172800000
      },
      "tags": [],
      "footer_offsets": {
        "start": 4714989,
        "end": 4719999
      },
      "delete_opstamp": 0,
      "num_merge_ops": 0
    }
  ]
}
```

Closes: #3907

### CODE REVIEW

File: quickwit/quickwit-serve/src/index_api/rest_handler.rs [hunk 1]
```diff
@@ -241,6 +241,14 @@ fn describe_index_handler(
 #[derive(Debug, Clone, Deserialize, Serialize, utoipa::IntoParams, utoipa::ToSchema, Default)]
 #[into_params(parameter_in = Query)]
 pub struct ListSplitsQueryParams {
+    /// If set, restrict the number of splits to skip
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/index_api/rest_handler.rs:244]:
```suggestion
    /// If set, define the number of splits to skip
```

File: quickwit/quickwit-serve/src/index_api/rest_handler.rs [hunk 2]
```diff
@@ -263,12 +271,22 @@ pub struct ListSplitsQueryParams {
...
 }
 
+#[derive(Serialize, Deserialize, Debug, utoipa::ToSchema)]
+pub struct ListSplitsResponse {
+    #[serde(default)]
+    pub offset: usize,
+    #[serde(default)]
+    pub size: usize,
+    #[serde(default)]
+    pub splits: Vec<Split>,
```
Comment (fmassot) [inline quickwit/quickwit-serve/src/index_api/rest_handler.rs:281]: We will probably use pagination for other resources: indexes for example. Replacing the field `splits` would be better.

**Review by fmassot — CHANGES_REQUESTED** [review]
I have only one minor comment, can you have a look at it?

File: quickwit/quickwit-serve/src/index_api/rest_handler.rs (hunk 2 above)
Comment (JerryKwan) [reply quickwit/quickwit-serve/src/index_api/rest_handler.rs:300]: Currently it's only used as the response of list split operation, that's why I use splits as the name. Maybe we can define different structures to to be used in different responses

**Review by fmassot — APPROVED** [review]

**Pull request merged**